### PR TITLE
Add parsing of case? statement and simple test.

### PR DESCRIPTION
**Pull request by bpadalino**
Also add the Matching flag to the Iir_Kind_Case_Statement.

**Description**
This adds the parsing for matching case statements (case?) as well as adds the Matching flag for the AST node.

### CODE REVIEW

File: src/vhdl/vhdl-parse.adb
Comment (umarcor) [inline src/vhdl/vhdl-parse.adb:7635]: Is `LRM93 10.9` correct?

Comment (bpadalino) [reply src/vhdl/vhdl-parse.adb]: Indeed it isn't.  Resolved.

File: testsuite/gna/issue717/matching_case.vhdl
Comment (umarcor) [inline testsuite/gna/issue717/matching_case.vhdl:16]: Maybe `CountStateM` can be of type unsigned, to avoid all the castings below?

File: pyGHDL/libghdl/vhdl/nodes.py
Comment (umarcor) [inline pyGHDL/libghdl/vhdl/nodes.py:1099]: Execute `./scripts/update_py_bindings.sh` right before committing your changes, in order to have Python sources regenerated. Then, run `black pyGHDL` to have consistent formatting. Black is a Python package, which you can install through pip.

File: testsuite/gna/issue717/matching_case.vhdl
Comment (bpadalino) [reply testsuite/gna/issue717/matching_case.vhdl]: Changed.  I mostly just took the example from 717.  There's definitely a lot that this doesn't test.  Just very basic.

File: pyGHDL/libghdl/vhdl/nodes.py
Comment (bpadalino) [reply pyGHDL/libghdl/vhdl/nodes.py:1099]: OK.  I believe I re-ran those and updated.  Let's see.

Comment (umarcor) [reply pyGHDL/libghdl/vhdl/nodes.py:1099]: It seems that you setup keeps removing some lines in `pyGHDL/libghdl/vhdl/nodes.py`. Can you check that black is up to date (CI uses `black-21.9b0`) and that no other editor/IDE settings are modifying the sources?

Comment (bpadalino) [reply pyGHDL/libghdl/vhdl/nodes.py]: OK, I'll keep this unresolved until we figure it out.

I think this latest push might have fixed?  Let's wait for the CI to tell us.

Comment (umarcor) [reply pyGHDL/libghdl/vhdl/nodes.py]: Yes, it did! Thanks!

**Review by umarcor — APPROVED** [review]
LGTM! I'll let @tgingold merge, though.

**Comment by tgingold** [comment]
Thanks!

**Pull request merged**
